.circleci/config.yml

```diff
@@ -117,7 +117,13 @@ jobs:
           no_output_timeout: 25m
           command: |
             export PYTHONPATH=$PWD/build-install/lib/python3.11/site-packages
-            python dev.py --no-build doc -j2
+            python dev.py --no-build doc -j2  2>&1 | tee sphinx_log.txt
+
+      - run:
+          name: Check sphinx log for warnings (which are treated as errors)
+          when: always
+          command: |
+            ! grep "^.* WARNING: .*$" sphinx_log.txt
```

Comment:
I adapted this [from MNE-Python](https://github.com/mne-tools/mne-python/blob/0a0cad8802e832669bb954a3bdd8e08bfaecf784/.circleci/config.yml#L301-L306) where we've used it for months, it helps make it clear what the warning line was so you don't have to ctrl-F it on CircleCI. In this PR's case it would have been:
```
/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/signal/_fir_filter_design.py:docstring of scipy.signal._fir_filter_design.minimum_phase:108: WARNING: Exception occurred in plotting scipy-signal-minimum_phase-1
```